pyexchange/kucoin.py

```diff
@@ -0,0 +1,269 @@
+
+    def get_balance(self, coin : str):
+        assert(isinstance(coin, str))
+        return self.client.get_coin_balance(coin)
+
+    def get_user_info(self):
+        return self.client.get_user()
+
+    def order_book(self, pair: str, limit=None):
+        assert(isinstance(pair, str))
+        return self.client.get_order_book(pair, limit)
+
+    def get_orders(self, pair: str)  -> List[Order]:
+        assert(isinstance(pair, str))
+
+        orders = self.client.get_active_orders(pair)
+
+        sell_orders = list(map(lambda item: Order.from_list(item, pair, True), orders[self.client.SIDE_SELL]))
+        buy_orders = list(map(lambda item: Order.from_list(item, pair, False), orders[self.client.SIDE_BUY]))
+
+        return sell_orders + buy_orders
+
+    def place_order(self, pair: str, is_sell: bool, price: Wad, amount: Wad) -> str:
+        assert(isinstance(pair, str))
+        assert(isinstance(is_sell, bool))
+        assert(isinstance(price, Wad))
+        assert(isinstance(amount, Wad))
+
+        side = self.client.SIDE_SELL if is_sell else self.client.SIDE_BUY
+
+        coins = pair.split("-")
+
+        price = self._get_precision(coins[1]) % float(price)
+        amount = self._get_precision(coins[0]) % float(price)
+
+        self.logger.info(f"Placing order ({side}, amount {amount} of {pair},"
+                         f" price {price})...")
+
+        result = self.client.create_order(pair, side, price, amount)
+
+        order_id = result['orderOid']
+
+        self.logger.info(f"Placed order as #{order_id}")
+
+        return order_id
+
+    def cancel_order(self, order_id: str, is_sell: bool, pair: str):
+        assert(isinstance(order_id, str))
+        assert(isinstance(is_sell, bool))
+        assert(isinstance(pair, str))
+
+        side = self.client.SIDE_SELL if is_sell else self.client.SIDE_BUY
+
+        self.logger.info(f"Cancelling order #{order_id} of type {side}...")
+
+        try:
+            self.client.cancel_order(order_id, side, pair)
+            self.logger.info(f"Canceled order #{order_id}...")
+            return True
+        except Exception as e:
+            self.logger.error(f"Failed to cancel order #{order_id}... {e}")
+            return False
+
+    def get_trades(self, pair: str, page_number: int = 1) -> List[Trade]:
+        assert(isinstance(pair, str))
+        assert(isinstance(page_number, int))
+
+        page_number = page_number - 1
+        limit = 100
+
+        result = self.client.get_symbol_dealt_orders(pair, page_number, limit)
+
+        return list(map(lambda item: Trade.from_dict(pair, item), result['datas']))
+
+    def get_all_trades(self, pair: str) -> List[Trade]:
+        assert(isinstance(pair, str))
+
+        result = self.client.get_recent_orders(pair)
+        print(result)
+
+        return list(map(lambda item: Trade.from_list(pair, item), result))
+
+    @staticmethod
+    def _get_precision(coin):
+        return {
```

Comment:
:/ is there no way to fetch it from the API dynamically..?

Reply:
didn't find such :(